src/transformers/onnx/config.py

```diff
@@ -502,7 +506,10 @@ def fill_with_past_key_values_(self, inputs_or_outputs: Mapping[str, Mapping[int
         name = "past_key_values" if direction == "inputs" else "present"
         for i in range(self.num_layers):
             inputs_or_outputs[f"{name}.{i}.key"] = {0: "batch", 2: "past_sequence + sequence"}
-            inputs_or_outputs[f"{name}.{i}.value"] = {0: "batch", 2: "past_sequence + sequence"}
+            if inverted_values_shape:
+                inputs_or_outputs[f"{name}.{i}.value"] = {0: "batch", 1: "past_sequence + sequence"}
```

Comment:
@lewtun Do you know how this string `past_sequence + sequence` is used? Is there some syntax we need to follow to make ONNX things work? For example, what happens if I write `past_sequence + present_sequence`? Or even `0: "batch_dim"`?

I have ONNX  knowledge only of measure 0 😢 Is there any some guide to understand this?

Reply:
Also, from
https://github.com/huggingface/transformers/blob/c8315e3bb0d612993983f401acf44b447e9280d3/src/transformers/onnx/config.py#507
(or just what @NouamaneTazi did in this PR), it looks like we need to specify dynamic axes not only for the model inputs , but **also required for all intermediate layers**. Is this correct?

Reply:
@ydshieh these strings refer to the names one assigns to nodes in the ONNX graph. It is just for readability, so changing the string has no effect on the API of `transformers.onnx`

See e.g. the PyTorch docs for more details: https://pytorch.org/docs/stable/onnx.html#example-alexnet-from-pytorch-to-onnx

Regarding your question about intermediate layers, these names are inferred from the ONNX ops generated in the export, so we never have to define them manually. For example, here's the names associated with BERT (directly from the Hub 🔥 ): https://netron.app/?url=https://huggingface.co/cmarkea/distilcamembert-base-ner/blob/main/model.onnx